### PR TITLE
Added OAUTH2_ACCESS_TOKEN_URL setting to discovery service

### PR DESCRIPTION
**Pull request by clintonb**
ECOM-3895

### CODE REVIEW

**Comment by clintonb** [comment]
@edx/devops please review

**Comment by feanil** [comment]
:+1:

**Comment by clintonb** [comment]
@jibsheet I updated this per our conversation earlier today. Would you review, please? I will trim this to 2 commits after approval.

**Comment by jibsheet** [comment]
@clintonb at first glance, looks fine, I want to grep in the secure repos when I'm on a machine with those checked out to make sure I know if we need a matching PR over there, or just deletions.  Will look in the morning.

**Comment by jibsheet** [comment]
:+1: 
@clintonb I need to get thumbs on the two private PRs, otherwise if we build out of sync, credentials or discovery will fail to log in (we should probably do test builds of them once everything merges).
I'll ping when mine are ready to merge.

**Comment by jibsheet** [comment]
@clintonb unfortunately, the travis failures are real, the CI jobs are failing because the docker containers don't have EDXAPP_LMS_ISSUER - you'll need to take a look at that code.

**Comment by clintonb** [comment]
@jibsheet ready for final review; I ended up adding a default to work with Docker.
@cpennington please review the changes to `docker.mk`.

File: docker.mk [hunk 1]
```diff
@@ -26,7 +26,7 @@ pkg: docker.pkg
 clean:
 	rm -rf .build
 
-docker.test.shard: $(foreach image,$(shell echo $(images) | tr ' ' '\n' | sed -n '$(SHARD)~$(SHARDS)p'),$(docker_test)$(image))
+docker.test.shard: $(foreach image,$(shell echo $(images) | tr ' ' '\n' | awk 'NR%$(SHARDS)==$(SHARD)'),$(docker_test)$(image))
```
Comment (cpennington) [inline docker.mk:29]: Ah, that's where that was coming from.

**Comment by cpennington** [comment]
:+1:

**Comment by jibsheet** [comment]
:+1: Assuming our builds check out in stage

**Comment by clintonb** [comment]
Validated on stage!

**Pull request merged**